### PR TITLE
Fix for non-ascii character in the path of backup engine (issue #339)

### PR DESCRIPTION
**Pull request by digglife**
#339

The basic idea is using bytes string under py2, using string(unicode) under py3. Although this fix is for Google Drive at the first, but it works for other engines as well. I also modified the test files a bit for testing the changes under both py2 and py3. For more details, please refer to the inline comments.

### CODE REVIEW

File: mackup/utils.py [hunk 1]
```diff
@@ -238,7 +241,7 @@ def get_google_drive_folder_location():
                      "WHERE entry_key = 'local_sync_root_path';")
             cur.execute(query)
             data = cur.fetchone()
-            googledrive_home = unicode(data[0])
+            googledrive_home = data[0]
```
Comment (digglife) [inline mackup/utils.py]: `unicode` is not a valid method under py3. It returns unicode string anyway.

File: mackup/config.py [hunk 2]
```diff
@@ -218,11 +218,10 @@ def _parse_path(self):
                 raise ConfigError("The required 'path' can't be found while"
                                   " the 'file_system' engine is used.")
 
-        # Python 2 and python 3 byte strings are different.
+        # For py2, encode the path with utf8 to bytes string because
+        # there might be non-ascii characters in it.
         if sys.version_info[0] < 3:
-            path = str(path)
-        else:
-            path = path.decode("utf-8")
```
Comment (digglife) [inline mackup/config.py]: under py3, `decode` method is only valid when path is `bytes` type, but path can be either `str` or `bytes` in the original code.

File: tests/config_tests.py [hunk 3]
```diff
@@ -9,6 +10,11 @@
                               ENGINE_FS)
 from mackup.config import Config, ConfigError
 
+def is_equal_string(src_path, dst_path):
+    if sys.version_info[0] < 3:
+        src_path = src_path.decode('utf8')
+    return src_path == dst_path
+
```
Comment (digglife) [inline tests/config_tests.py]: `decode` the bytes string path for comparing with unicode string in tests(py2).

File: mackup/utils.py [hunk 4]
```diff
@@ -206,7 +209,7 @@ def get_dropbox_folder_location():
             data = f_hostdb.read().split()
     except IOError:
         error("Unable to find your Dropbox install =(")
-    dropbox_home = base64.b64decode(data[1])
+    dropbox_home = base64.b64decode(data[1]).decode('utf8')
```
Comment (digglife) [inline mackup/utils.py]: `base64.b64decode(data[1])` returns bytes under py3. `decode` here for getting a unicode string, no matter the environmet is  py2 or py3.

**Comment by rposborne** [comment]
This is causing some issues as it will prevent someone from uninstalling mackup, could we get this rebased on master and merged?

**Pull request closed without merging**